Added evaluate and tap commands last-sexp and sexp-at-point

skjoenberg opened the pull request:
This PR solves #3239 by adding two commands that evaluate and tap either the last-sexp or the sexp-at-point. 
This is simply done by wrapping the form in a `(doto *form* (tap>))`.

I've not added any tests, since I couldn't find much in the `cider-eval-tests.el`. I'll happily do it if anybody can point me in the right direction.

I've not chosen any keybinding for this. I don't have any clue on how to pick them.

-----------------

Before submitting the PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines](../blob/master/.github/CONTRIBUTING.md)
- [ ] You've added tests (if possible) to cover your change(s)
- [x] All tests are passing (`eldev test`)
- [x] All code passes the linter (`eldev lint`) which is based on [`elisp-lint`](https://github.com/gonewest818/elisp-lint) and includes
  - [byte-compilation](https://www.gnu.org/software/emacs/manual/html_node/elisp/Byte-Compilation.html), [`checkdoc`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Tips.html), [check-declare](https://www.gnu.org/software/emacs/manual/html_node/elisp/Declaring-Functions.html), packaging metadata, indentation, and trailing whitespace checks.
- [x] You've updated the [changelog](../blob/master/CHANGELOG.md) (if adding/changing user-visible functionality)
- [ ] You've updated the [user manual](../blob/master/doc) (if adding/changing user-visible functionality)

Thanks!

*If you're just starting out to hack on CIDER you might find this [section of its
manual][1] extremely useful.*

[1]: https://docs.cider.mx/cider/contributing/hacking.html

vemv commented on cider-eval.el:
```diff
@@ -1016,6 +1016,26 @@ If invoked with OUTPUT-TO-CURRENT-BUFFER, output the result to current buffer."
+buffer."
+  (interactive "P")
+  (let* ((form (apply #'buffer-substring-no-properties (cider-last-sexp 'bounds)))
+         (tapped-form (concat "(doto " form " (tap>))")))
```
```suggestion
         (tapped-form (concat "(clojure.core/doto " form " (clojure.core/tap>))")))
```

We never know if something is shadowing them

skjoenberg replied:
Ah, yes! Haven't thought of that, thanks.

vemv reviewed (commented):
This sounds a reasonable addition to me, that might be handy to others

bbatsov commented on cider-eval.el:
```diff
@@ -1016,6 +1016,26 @@ If invoked with OUTPUT-TO-CURRENT-BUFFER, output the result to current buffer."
+If invoked with OUTPUT-TO-CURRENT-BUFFER, print the result in the current
+buffer."
+  (interactive "P")
+  (let* ((form (apply #'buffer-substring-no-properties (cider-last-sexp 'bounds)))
```
I'm surprised `cider-last-sexp` would returns something with unstripped properties. Probably something like this should be done there to avoid repetition in the client code.

skjoenberg replied:
I found that `cider-last-exp` actually did this, so I just used it directly without `'bounds`. I hope this fixes your comments, otherwise, I might need a bit more explanation.

bbatsov commented:
New eval commands should be added to both the menu and the eval keymap to ease their discovery.

skjoenberg commented:
> New eval commands should be added to both the menu and the eval keymap to ease their discovery.

I've added two keybindings and added the commands to `cider-mode-eval-menu`. If this was not the menu you were referring to, I'll need a hint to a file. :)

bbatsov commented:
Looks good. Thanks!
